app/screens/component_library/button.cl.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {Alert, View} from 'react-native';
+
+import Button from '@components/button';
+import {useTheme} from '@context/theme';
+
+import {useBooleanProp, useDropdownProp, useStringProp} from './hooks';
+import {buildComponent} from './utils';
+
+const propPossibilities = {};
+
+const buttonSizeValues = ['xs', 's', 'm', 'lg'];
+const buttonEmphasisValues = ['primary', 'secondary', 'tertiary', 'link'];
+const buttonTypeValues = ['default', 'destructive', 'inverted', 'disabled'];
+const buttonStateValues = ['default', 'hover', 'active', 'focus'];
```

Comment:
I wonder if there is a way for you to construct this array of values based on the styles of the buttons defined in `@utils/buttonStyles` so that if ever another one is added it won't be missing from here.

Reply:
Sadly, it is a limitation from typescript. We can get types from runtime objects, but we cannot get runtime objects from types :sweat_smile:
There are ways around that, but all of them look pretty ugly to me, and would imply extra changes in the code: https://stackoverflow.com/questions/75835161/typescript-convert-type-with-multiple-string-options-to-an-array-with-those-opti

Nevertheless, it is something we can consider for the future.